daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2126,9 +2157,11 @@ func (devices *DeviceSet) removeDevice(devname string) error {
 	return err
 }
 
-func (devices *DeviceSet) cancelDeferredRemoval(info *devInfo) error {
+func (devices *DeviceSet) cancelDeferredRemoval(info *devInfo) (bool, error) {
```

Comment:
Basically we need two functions.

- One which just ensures that when the function finishes, device should not be in deferred remove state. Other states like device not being there any more etc are fine. This is what current function is implementing.

- Another function which returns success only if deferred removal was cancelled otherwise appropriate error message. 

Shishir, lets implement this second variant of function and any other required bits let open code in the caller. And that should take care of concern raised by @cpuguy83